src/main/java/com/opencart/steps/SearchProductPageBL.java

```diff
@@ -0,0 +1,41 @@
+package com.opencart.steps;
+
+import com.opencart.pages.SearchProductPage;
+import com.opencart.pages.SuccessSearchProductResultPage;
+import org.testng.Assert;
+
+/*
+* Class that describes steps (Business logic) of searching product
+* with correct search criteria "mac"
+ */
+
+public class SearchProductPageBL {
+
+    private SearchProductPage searchProductPage;
+    private SuccessSearchProductResultPage successSearchProductResultPage;
+
+    public SearchProductPageBL() {
+        searchProductPage = new SearchProductPage();
+    }
+
+    public SearchProductPageBL makeSearch() {
+        searchProductPage.getInputSearchCriteria().clear();
+        searchProductPage.getInputSearchCriteria().sendKeys("mac");
+        clickOnContinueButton();
+        successSearchProductResultPage = new SuccessSearchProductResultPage();
+        return this;
+    }
+
+    public void verifySearchResult() {
```

Comment:
Try to make additional validation of search. For example, if search is successful, verify the list of products displayed. So, each product should contains your search phrase.